python/pyarrow/tensorflow/plasma_op.cc

```diff
@@ -350,6 +350,10 @@ class PlasmaToTensorOp : public AsyncOpKernel {
  private:
   std::string plasma_store_socket_name_;
   std::string plasma_manager_socket_name_;
+
+  mutex mu_;
+  bool connected_ = false;
+  plasma::PlasmaClient client_ GUARDED_BY(mu_);
```

Comment:
This is one plasma client per custom op? The challenge with that is you could easily need hundreds or thousands of custom ops in a single TF graph, which would be a lot of plasma clients, potentially running out of file descriptors somewhere.

We can certainly try it this way and see what happens.

Reply:
What do you mean there is one op per graph. Couldn't there be an arbitrary number of ops per graph?

Reply:
In principle yes, but one per graph is how we are using it, I think it's fine to have one plasma client per op (and it's cleaner code wise since we don't need global variables in this case)